spec/models/vm_retire_task_spec.rb

```diff
@@ -0,0 +1,38 @@
+describe VmRetireTask do
+  let(:user) { FactoryGirl.create(:user_with_group) }
+  let(:vm) { FactoryGirl.create(:vm) }
+  let(:miq_request) { FactoryGirl.create(:vm_retire_request, :requester => user) }
+  let(:miq_request_task) { FactoryGirl.create(:miq_request_task, :miq_request_id => miq_request.id) }
+  let(:vm_retire_task) { FactoryGirl.create(:vm_retire_task, :source => vm, :miq_request_task_id => miq_request_task.id, :miq_request_id => miq_request.id, :options => {:src_ids => [vm.id] }) }
+  let(:reason) { "Why Not?" }
+  let(:approver) { FactoryGirl.create(:user_miq_request_approver) }
+  let(:zone) { FactoryGirl.create(:zone, :name => "fred") }
+
+  it "should initialize properly" do
+    expect(vm_retire_task.state).to eq('pending')
+    expect(vm_retire_task.status).to eq('Ok')
+  end
+
+  describe "deliver_to_automate" do
+    before do
+      allow(MiqServer).to receive(:my_zone).and_return(Zone.seed.name)
+      miq_request.approve(approver, reason)
+    end
+
+    it "calls set_automation_attributes_from_objects" do
+      expect(MiqAeEngine).to receive(:set_automation_attributes_from_objects).with(vm, "request"=>"vm_retire").and_return a_hash_including("Vm::vm"=>vm.id)
```

Comment:
Does this work? `a_hash_including("Vm::vm"=>vm.id)`  does it return a matcher rather than a hash?